patches/minecraft/net/minecraft/world/entity/monster/Phantom.java.patch

```diff
@@ -0,0 +1,51 @@
+--- a/net/minecraft/world/entity/monster/Phantom.java
++++ b/net/minecraft/world/entity/monster/Phantom.java
+@@ -152,6 +_,7 @@
+       }
+ 
+       this.m_33108_(p_33132_.m_128451_("Size"));
++      this.spawnCause = p_33132_.m_128441_("ForgeSpawnCause") ? p_33132_.m_128342_("ForgeSpawnCause") : null;
+    }
+ 
+    public void m_7380_(CompoundTag p_33141_) {
+@@ -160,6 +_,7 @@
+       p_33141_.m_128405_("AY", this.f_33098_.m_123342_());
+       p_33141_.m_128405_("AZ", this.f_33098_.m_123343_());
+       p_33141_.m_128405_("Size", this.m_33172_());
++      if (spawnCause != null) p_33141_.m_128362_("ForgeSpawnCause", this.spawnCause);
+    }
+ 
+    public boolean m_6783_(double p_33107_) {
+@@ -487,5 +_,32 @@
+ 
+          }
+       }
++   }
++
++   // Forge start
++   @Nullable
++   private java.util.UUID spawnCause = null;
++
++   /**
++    * @return UUID describing what caused phantom to spawn (generally a player's UUID). null if not applicable.
++    */
++   @Nullable
++   public java.util.UUID getSpawnCause()
++   {
++      return this.spawnCause;
++   }
++
++   /**
++    * Allows to specify spawn cause for phantoms (for event listeners).<br>
++    * <br>
++    * This should be called by mods right away when they create phantoms, so
++    * event listeners such as {@link net.minecraftforge.event.entity.EntityJoinLevelEvent} can figure out
++    * what caused this phantom to spawn.<br>
++    *
++    * @param spawnCause UUID describing what caused phantom to spawn (generally a player's UUID).
++    */
++   public void setSpawnCause(@Nullable java.util.UUID spawnCause)
++   {
++      this.spawnCause = spawnCause;
```

Comment:
This should be guarded to not allow changing the value once it has been set.  Specifically, it should throw `UnsupportedOperationException` if `this.spawnCause` is not null.